overlord/hookstate/hookmgr.go

```diff
@@ -328,44 +324,33 @@ func runHookAndWait(snapName string, revision snap.Revision, hookName, hookConte
 	}()
 
 	var abortOrTimeoutError error
-	dyingCh := tomb.Dying()
-out:
-	for {
-		select {
+	select {
+	case <-hookCompleted:
 		// Hook completed; it may or may not have been successful.
-		case <-hookCompleted:
-			break out
-		// Max timeout reached, process will get killed below
-		case <-killTimerCh:
-			abortOrTimeoutError = fmt.Errorf("exceeded maximum runtime of %s", maxRuntime)
+		return buffer.Bytes(), hookError
+	case <-tomb.Dying():
 		// Hook was aborted, process will get killed below
-		case <-dyingCh:
-			dyingCh = nil
-			abortOrTimeoutError = fmt.Errorf("hook %q aborted", hookName)
+		abortOrTimeoutError = fmt.Errorf("hook %q aborted", hookName)
+	case <-killTimerCh:
+		// Max timeout reached, process will get killed below
+		abortOrTimeoutError = fmt.Errorf("exceeded maximum runtime of %s", timeout)
```

Comment:
The hook-name was initially there :) But the returned error is used to construct a longer error message in https://github.com/mvo5/snappy/blob/4b21aed97c0cd38d0586f611f06c8c64844d38d8/overlord/hookstate/hookmgr.go#L208 so the hook-name would be duplicated. I can rework the way the errors are constructed if this feels awkward.